js/foundation.abide.js

```diff
@@ -428,13 +442,6 @@ class Abide extends Plugin {
       else if (pattern !== $el.attr('type')) {
         valid = new RegExp(pattern).test(inputText);
       }
-      else {
```

Comment:
This sees to be unrelated.

Reply:
The `validateText` method was performing two validations:
1. pattern validation
2. checked if the input is required

It returns false if either of the two validations fails, but to implement this feature, I need to distinguish between those two cases. Please see:
https://github.com/foundation/foundation-sites/pull/11698/files#diff-37998d120faf57b3e7580109c8e306e2R346-R347

I think this is a reasonable change because:
1. the method now does only one thing instead of two
2. the documentation now matches the implementation